### DISCUSSION TITLE
Kicking tasks from a task in another broker.

### DISCUSSION THREAD

**Discussion by lembregtse** [taskiq-python/taskiq#80] [Q&A] [answered]
Quick question, we are using aio_pika brokers with several tasks each. Sometimes we have a requirement where one task kicks off another task in another broker. 

Because of this the startup method of the other brokers is never executed (due to it being in the main code).

So we receive the obvious error "ValueError: Please run startup before kicking."

What would be the recommended approach here?

**Comment by s3rius** [comment, marked as answer]
I guess, you can add startup of another broker in your primary one. 

Like this:
```python
from taskiq import InMemoryBroker, TaskiqEvents, TaskiqState

broker1 = InMemoryBroker()
broker2 = InMemoryBroker()


@broker1.on_event(TaskiqEvents.WORKER_STARTUP)
async def run_another_broker(_: TaskiqState):
    await broker2.startup()
```

Replace inmemory brokers with your own.

**Reply by s3rius** [reply]
Because when you start 

```
taskiq worker script:broker1
```

It will automatically call startup for the specified broker.

**Reply by s3rius** [reply]
If that works for you, mark this response as answer.
